Port `display.c` to SDL3

ankith26 opened the pull request:
WIP to port `display.c` to SDL3

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- New Features
  - SDL3 compatibility for display and window management.
  - Updated display enumeration, mode queries, and refresh rate reporting.
  - Unified integer scaling behavior across SDL2 and SDL3.
  - Improved window/WM info retrieval and fullscreen handling.

- Bug Fixes
  - More reliable fullscreen transitions and state detection across platforms.
  - Safer pixel format handling.

- Refactor
  - Broad internal updates to support dual SDL2/SDL3 code paths without changing public APIs.

- Chores
  - Display module is now always built.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ankith26 commented:
This PR should be done from my side.

Given that this is a kinda big PR to review, I have already organized everything into commits that take up stuff one at a time and focus on it fully. So, if it helps reviewing I can split this up into multiple PRs each with subsets of commits from here.

damusss commented on src_c/window.c:
```diff
@@ -331,6 +331,10 @@ pg_window_set_fullscreen(SDL_Window *window, int desktop)
             goto end;
         }
     }
+
```
What is this doing?

ankith26 replied:
Basically while working on fullscreen changes in display I saw that `SDL_SetWindowFullscreenMode` in SDL3 has effect only if the window is already in fullscreen mode according to the docs. So I added a call here as well that does that first.

Starbuck5 commented on src_c/display.c:
```diff
@@ -2787,31 +3476,27 @@ pg_display_resize_event(PyObject *self, PyObject *event)
         return PyLong_FromLong(-1);
     }
 
-    // could also take the size of the old display surface
-    SDL_GetWindowSize(win, &window_w, &window_h);
-    window_display = SDL_GetWindowDisplayIndex(win);
-    if (SDL_GetDesktopDisplayMode(window_display, &display_mode) != 0) {
-        return RAISE(pgExc_SDLError, SDL_GetError());
-    }
-
```
This was just unused?

Starbuck5 reviewed (approved):
Thanks Ankith! I looked through the changes and it seems good.

This did take a significant amount of time to read through, maybe if some of the smaller commits were split into their own PRs they could be reviewed more quickly by others? Not sure the best move.

When I combine with my surface branch I'm unable to open a window, but this PR is for compiling and not necessarily for getting everything to run.

```
>>> screen = pygame.display.set_mode((500, 500))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
pygame.error: Invalid display
```

ankith26 commented:
I rebased this PR on main to get the latest surface compilation changes (hence the force push). While I was at it I also fixed a super minor return code related issue in an SDL3 specific part of this PR, and as a result `set_mode` now works on SDL3! * 

*: `set_mode` on SDL3 only works when called after `pygame.init()` as of now. This can be fixed but needs a bit more code restructuring that would also affect SDL2 paths, hence I am keeping that out of this PR.

With this PR we have some examples in a somewhat usable/runnable state on SDL3

ankith26 commented on src_c/display.c:
```diff
@@ -427,15 +450,28 @@ pg_GetVideoInfo(pg_VideoInfo *info)
         info->current_w = winsurf->w;
         info->current_h = winsurf->h;
-        info->vfmt_data = *(winsurf->format);
+        PG_PixelFormat *fmt = PG_GetSurfaceFormat(winsurf);
+        if (!fmt) {
+            PyErr_SetString(pgExc_SDLError, SDL_GetError());
+            return (pg_VideoInfo *)NULL;
+        }
+        info->vfmt_data = *fmt;
         info->vfmt = &info->vfmt_data;
```
no
